libretroshare/src/util/rsfile.cc

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * libretroshare/src/util: rsfile.cc                                           *
+ *                                                                             *
+ * libretroshare: retroshare core library                                      *
+ *                                                                             *
+ * Copyright (C) 2021 Retroshare Team <retroshare.project@gmail.com>           *
+ *                                                                             *
+ * This program is free software: you can redistribute it and/or modify        *
+ * it under the terms of the GNU Lesser General Public License as              *
+ * published by the Free Software Foundation, either version 3 of the          *
+ * License, or (at your option) any later version.                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
+ * GNU Lesser General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Lesser General Public License    *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
+ *                                                                             *
+ *******************************************************************************/
+
+#include "util/rsfile.h"
+
+#ifdef WINDOWS_SYS
+#include <wtypes.h>
+#include <io.h>
+#include <namedpipeapi.h>
+#include <errno.h>
+#else
+#include <fcntl.h>
+#endif
+
+int RsFileUtil::set_fd_nonblock(int fd)
```

Comment:
why did you need to create this function? unix_fcntl_nonblock() does this already (for sockets), but it should only be called for sockets anyway.

Reply:
I thought unix_fcntl_nonblock can be used for file descriptors and sockets on Linux. A call to unix_fcntl_nonblock was already there in TorProcess::start for stdout and stderr.
On Windows there are two different functions to set a fd to non blocking. The Windows specific part in unix_fcntl_nonblock can set a socket to non blocking and the Windows specific part in RsFileUtil::set_fd_nonblock can set a file descriptor like stdout and stderr to non blocking.
The Linux specific part is the same in both functions. Maybe one function can call the other to not have the same code twice.